cinema/tests/test_movie_viewset.py

```diff
@@ -0,0 +1,162 @@
+        self.assertEqual(len(response.data), 1)
+        self.assertEqual(response.data[0]["title"], "Movie One")
+
+    def test_create_movie_as_authenticated_user(self):
+        url = reverse("cinema:movie-list")
+        data = {
+            "title": "New Movie",
+            "description": "A new movie description",
+            "duration": 90,
+            "genres": [self.genre1.id],
+            "actors": [self.actor1.id, self.actor2.id],
+        }
+        response = self.client.post(url, data, format="json")
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+        expected_error = {
+            "detail": ErrorDetail(
+            string="You do not have permission to perform this action.",
+            code="permission_denied"
+            )
+        }
+        self.assertEqual(response.data, expected_error)
+
+    def test_create_movie_as_admin(self):
+        self.client.credentials(HTTP_AUTHORIZATION=f"Token {self.admin_token.key}")
+
+        url = reverse("cinema:movie-list")
+        data = {
+            "title": "New Movie",
+            "description": "A new movie description",
+            "duration": 90,
+            "genres": [self.genre1.id],
+            "actors": [self.actor1.id, self.actor2.id],
+        }
+
+        response = self.client.post(url, data, format="json")
+
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+
+        self.assertEqual(response.data["title"], "New Movie")
+
+        movie = Movie.objects.get(title="New Movie")
+
+        self.assertEqual(movie.description, "A new movie description")
+        self.assertEqual(movie.duration, 90)
+        self.assertEqual(list(movie.genres.values_list('id', flat=True)), [self.genre1.id])
+        self.assertEqual(list(movie.actors.values_list('id', flat=True)), [self.actor1.id, self.actor2.id])
+
+    def test_retrieve_movie(self):
+        url = reverse("cinema:movie-detail", args=[self.movie1.id])
+        response = self.client.get(url)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response.data["title"], "Movie One")
+        response_genre_ids = [genre['id'] for genre in response.data["genres"]]
+        expected_genre_ids = [self.genre1.id]
+        self.assertEqual(response_genre_ids, expected_genre_ids)
+        response_actor_ids = [actor['id'] for actor in response.data["actors"]]
+        expected_actor_ids = [self.actor1.id]
+        self.assertEqual(response_actor_ids, expected_actor_ids)
+
+    def test_upload_movie_image_as_authenticated_user(self):
+        url = reverse(
+            "cinema:movie-upload-image", args=[self.movie1.id]
+        )  # Updated URL name
+        image_file = generate_image_file()
+        response = self.client.post(url, {"image": image_file}, format="multipart")
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+    def test_authenticated_user_cannot_delete_movie(self):
```

Comment:
This test checks if a regular authenticated user can delete a movie and expects a 403 Forbidden response. This is a good test to have. However, consider also adding a test that checks if an unauthenticated user can delete a movie. This would be another important test to have to ensure that unauthenticated users cannot delete movies.